src/ray/raylet/node_manager.cc

```diff
@@ -763,6 +763,11 @@ void NodeManager::ProcessClientMessage(
     ProcessRegisterClientRequestMessage(client, message_data);
   } break;
   case protocol::MessageType::GetTask: {
+    RAY_CHECK(!registered_worker->UsePush());
+    ProcessGetTaskMessage(client);
+  } break;
+  case protocol::MessageType::TaskDone: {
+    RAY_CHECK(registered_worker->UsePush());
```

Comment:
I'm confused about the difference between this and `GetTask`? When do we call one and/or the other?

Reply:
The rule is simple: 
- for legacy workers they still use `GetTask`, which indicates previous task is done and fetches a new task from raylet; 
- for workers based on c++ core worker, they will use `TaskDone`, just to indicate a task is finished, and then raylet will push a new task to the worker via grpc.